### PR TITLE
Add external links

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: src/support.sch [hunk 1]
```diff
@@ -303,6 +304,30 @@
         </rule>
     </pattern>
 
+    <pattern id="ext-link">
+        <rule context="ext-link[not(contains(@xlink:href, ':') or ancestor-or-self::*[contains(@xml:base, ':')])]" role="warn">
```
Comment (thewilkybarkid) [inline src/support.sch:308]: Not correct, but think it's the best we can do with XPath 1...

File: tests/cases/ext-link/contents.xml [hunk 2]
```diff
@@ -2,10 +2,10 @@
 <?xml-model href="../../jats.rng" schematypens="http://relaxng.org/ns/structure/1.0"?>
 <?xml-model href="../../../src/support.sch" schematypens="http://purl.oclc.org/dsdl/schematron"?>
 
-<aff>
+<ext-link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="http://example.com/">
```
Comment (thewilkybarkid) [inline tests/cases/ext-link/contents.xml:5]: Renamed `inline.xml`s to be consistently `contents.xml`, which has caused the odd diff here.

File: src/support.sch (hunk 1 above)
Comment (fred-atherden) [reply src/support.sch:308]: XPath 1 has a function `starts-with($variable,$test)`. We could use this to mandate that only links that begin with a known uri scheme are supported, example:
```xml
<rule context="ext-link" role="warn">
            <assert test="starts-with(@xlink:href,'https://') or starts-with(@xlink:href,'http://') or starts-with(@xlink:href,'ftp://') or starts-with(@xlink:href,'mailto:') or starts-with(@xlink:href,'tel:') or starts-with(@xlink:href,'git://')">
                &lt;<name/>&gt; without an absolute @xlink:href is ignored.
            </assert>
        </rule>
```

Presumably there are hundreds of valid uri schemes, but if we were to output a message for the most used uri schemes, which again I assume is a handful, then we could cover the majority of cases.

Comment (thewilkybarkid) [reply src/support.sch:308]: I'll leave for now, something we can maybe revisit (don't want to block potential but obscure schemes).

**Review by fred-atherden — APPROVED** [review]

**Pull request merged**